simple_history/manager.py

```diff
@@ -9,13 +9,114 @@
+        type's primary key, and not the history_id (historical record's
+        primary key); this happens frequently with DRF.
+        """
+        if self._as_instances and "pk" in kwargs:
+            kwargs[self._pk_attr] = kwargs.pop("pk")
+        return super().filter(*args, **kwargs)
+
+    def latest_of_each(self):
+        """
+        Ensures results in the queryset are the latest historical record for each
+        primary key.  Deletions are not removed.
+
+        Returns a queryset.
+        """
+        # If using MySQL, need to get a list of IDs in memory and then use them for the
+        # second query.
+        # Does mean two loops through the DB to get the full set, but still a speed
+        # improvement.
+        backend = connection.vendor
+        if backend == "mysql":
+            history_ids = {}
+            for item in self.order_by("-history_date", "-pk"):
+                if getattr(item, self._pk_attr) not in history_ids:
+                    history_ids[getattr(item, self._pk_attr)] = item.pk
+            latest_historics = self.filter(history_id__in=history_ids.values())
+        elif backend == "postgresql":
+            latest_pk_attr_historic_ids = (
+                self.order_by(self._pk_attr, "-history_date", "-pk")
+                .distinct(self._pk_attr)
+                .values_list("pk", flat=True)
+            )
+            latest_historics = self.filter(history_id__in=latest_pk_attr_historic_ids)
+        else:
+            latest_pk_attr_historic_ids = (
+                self.filter(**{self._pk_attr: OuterRef(self._pk_attr)})
+                .order_by("-history_date", "-pk")
+                .values("pk")[:1]
+            )
+            latest_historics = self.filter(
+                history_id__in=Subquery(latest_pk_attr_historic_ids)
+            )
+        return latest_historics
+
+    def _clone(self):
+        c = super()._clone()
+        c._as_instances = self._as_instances
+        c._pk_attr = self._pk_attr
+        return c
+
+    def _fetch_all(self):
+        super()._fetch_all()
+        self._instanceize()
+
+    def _instanceize(self):
+        """
+        Convert the result cache to instances if possible and it has not already been
+        done.  If a query extracts `.values(...)` then the result cache will not contain
+        historical objects to be converted.
+        """
+        if (
+            self._result_cache
+            and self._as_instances
+            and isinstance(self._result_cache[0], self.model)
+        ):
+            self._result_cache = [item.instance for item in self._result_cache]
+
+
 class HistoryDescriptor:
     def __init__(self, model):
         self.model = model
 
     def __get__(self, instance, owner):
         if instance is None:
-            return HistoryManager(self.model)
+            return HistoryManager.from_queryset(HistoricalQuerySet)(self.model)
         return HistoryManager(self.model, instance)
```

Comment:
How come we don't want to use the `from_queryset` when instantiating the manager with an instance?

Reply:
Good question, I believe it would be useless to get a historical queryset when instantiating from an instance because HistoricalQuerySet exists to facilitate being able to query history or instances; if you are already at an instance then you do not need to query history.  That said, the better answer here is "I don't know" as my implementation and tests haven't needed it.  as_of queries in all my use cases are always on an model and not on an instance.